### PR TITLE
[20 min task but took 40] [Tagged metrics Part 3] Cassandra client pool metrics are now tagged

### PR DESCRIPTION
**Pull request by hsaraogi**
**Goals (and why)**: Tagged metrics for C* client pool. Partial step towards #2341. 

**Implementation Description (bullets)**: Get rid of pool number and add hostname as tags.

**Concerns (what feedback would you like?)**: NA

**Where should we start reviewing?**: CCPC.java

**Priority (whenever / two weeks / yesterday)**: soon.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2659)

### CODE REVIEW

File: atlasdb-client/src/main/java/com/palantir/atlasdb/util/MetricsManager.java [hunk 1]
```diff
@@ -65,6 +70,14 @@ public void registerMetric(Class clazz, String metricName, Metric metric) {
         registerMetricWithFqn(MetricRegistry.name(clazz, metricName), metric);
     }
 
+    public void registerTaggedGauge(Class clazz, String metricName, Map<String, String> tag, Gauge gauge) {
```
Comment (tboam) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/util/MetricsManager.java]: If we move the `Map<String, String> tag` to last argument and change the method name to just `registerMetric` then we can just go around adding tags to the end of anywhere we currently register metrics and so minimise the effort to add tags.

**Comment by hsaraogi** [comment]
Tritium tagged metrics does not support removing/deregistering metrics. We need to think about how we want to handle the cleanup here. @schlosna for thoughts?

**Comment by schlosna** [comment]
@hsaraogi yeah, we likely need a way to replace metrics in the TaggedMetricRegistry similar to https://github.com/palantir/tritium/blob/develop/tritium-metrics/src/main/java/com/palantir/tritium/metrics/MetricRegistries.java#L188

**Comment by hsaraogi** [comment]
Is that feature in the tritium pipeline?

**Comment by schlosna** [comment]
Open to PRs, I won't be able get to it this week.

@ellisjoe for SA

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java [hunk 2]
```diff
@@ -262,28 +262,27 @@ public String toString() {
...
-        return "pool" + poolNumber + "." + metric;
-    }
-
-    private void registerMetric(String metricName, Gauge gauge) {
-        metricsManager.registerMetric(CassandraClientPoolingContainer.class, metricName, gauge);
+    private void registerMetric(String metricName, String hostName, Gauge gauge) {
+        metricsManager.registerMetric(CassandraClientPoolingContainer.class,
+                metricName,
+                gauge,
+                ImmutableMap.of("cassandra-host", hostName));
```
Comment (ellisjoe) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java]: as per my email, I'd recommend holding of on tagging for the moment.

**Comment by fsamuel-bs** [comment]
Also worth checking if the fixes on #2662 are still valid with this PR.

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java (hunk 2 above)
Comment (ellisjoe) [reply atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java]: following up here. I don't think we want to tag the host. we've removed similar tags from remoting already: https://github.com/palantir/http-remoting/pull/642

**Comment by hsaraogi** [comment]
Closing in favour of #2786

**Pull request closed without merging**